pkg/datapath/linux/probes/probes.go

```diff
@@ -487,3 +465,149 @@ func HaveV3ISA() error {
 	}
 	return nil
 }
+
+// CreateHeaderFiles creates C header files with macros indicating which BPF
+// features are available in the kernel.
+func CreateHeaderFiles(headerDir string, probes *FeatureProbes) error {
```

Comment:
The code for creating the header files is three times the same, maybe factor those?

Reply:
Not sure what you mean, the three `writeXYZHeader()` are wrappers around `writeFeatureHeader()` to prepare arguments. What can be changed short of eliminating them and lifting all the logic into `CreateHeaderFiles()`?

Reply:
I think Quentin's suggestion makes sense. We could also have all error messages follow the same format with the filename to simplify things further (no `errMsg`).

Reply:
Or use `headerName` instead to differentiate the messages

Reply:
I don't think the `writerFunction()` suggestion above makes things any clearer and doesn't solve a problem. All function bodies still need to live somewhere, no need to over-DRY things.

As a follow-up, we could switch back to input structs (this is how the implementation started out), but this time around, bind macro names to struct fields using annotations:
```go
type CommonFeatures struct {
	LRUHashMap bool `macro:"HAVE_LRU_HASH_MAP_TYPE"`
	LPMTrieMap bool `macro:"HAVE_LPM_TRIE_MAP_TYPE"`
	...
}
```

This takes `FeatureProbes` out of the equation, which is shorter and much less prone to human error.

For now, I think this patch is clear enough to be merged, minus some nits that were brought up here.